Add Boss Fight Tutorial

CommanderFoo opened the pull request:
Doing things a little different compared to the race timer tutorial. This time, I have pushed the structure of it now so it can be reviewed first before I write all of the sections.

Ignore typos, wanted to push this now so I can get to writing it once the structure is approved.

CommanderFoo committed:
Add Boss Tutorial Outline Draft

Doing things a little different compared to the race timer tutorial. Pushing this now to get feedback on the structure of it.

CommanderFoo committed:
Added nav link

CommanderFoo committed:
Player needs a gun at the start to test floor tiles

PopThosePringles commented:
Don't worry about reviewing this now. It's changed a bit as I've been working on the tutorial. It's mostly the same, just tweaked a little, ordering changed etc.

CommanderFoo committed:
Add Boss Fight Tutorial

github-actions[bot] committed:
Optimised images with calibre/image-actions

CommanderFoo committed:
Changed all text the user needs to set to inline code

coreslinkous commented on src/tutorials/boss_tutorial.md:
```diff
@@ -0,0 +1,4330 @@
+
+1. Create a new script called **PlayerServer**.
+
+    Place the **PlayerServer** script inside the group **Boss Fight Map** in the **Hierarchy**
```
Make these subsets part of the larger numbered list. The "you can" sounds like it's optional, and really all of this is necessary to do in order.

coreslinkous commented on src/tutorials/boss_tutorial.md:
```diff
@@ -0,0 +1,4330 @@
+
+    ![!Damageable Object](../img/BossTutorial/damageable_object_core_content.png)
+
+2. Set the **Max Hit Points** to 1000.
```
Mention that this is in the **Properties** menu the first time.

coreslinkous commented on src/tutorials/boss_tutorial.md:
```diff
@@ -0,0 +1,4330 @@
+
+## Create AI Activity Handler
+
+In this section you will be create the AI for the boss. The boss needs to transition between different states depending on what is happening in the game. You will use an **AI Activity Handler** that will have 2 activities for handling the idle and shoot states.
```
Link to AI Activity System ref

coreslinkous commented on src/tutorials/boss_tutorial.md:
```diff
@@ -0,0 +1,4330 @@
+
+    Place the script **BossAIServer** as a child of **Boss AI Handler** in the **Hierarchy**.
+
+2. Drag **Boss Geo** onto the **BossAIServer** script as a custom property.
```
Swap "drag" to "add" to avoid ambiguity with adding it as a child.

coreslinkous commented on src/tutorials/boss_tutorial.md:
```diff
@@ -0,0 +1,4330 @@
+
+You need to create an idle activity and register that activity to the **Boss AI Handler**.
+
+1. Create activities table.
```
Convert to ol to h4 if nesting code blocks between steps

coreslinkous commented on src/tutorials/boss_tutorial.md:
```diff
@@ -0,0 +1,4330 @@
+
+You will be creating a shoot activity, and adding that activity to the handler.
+
+1. Add `shoot` table to `activities` table.
```
Same as above.

coreslinkous commented on src/tutorials/boss_tutorial.md:
```diff
@@ -0,0 +1,4330 @@
+
+## Create Boss Projectile
+
+The boss will be shooting a projectile at the current target. Players who get hit by the projectile will take damage.
```
Optional, but a static image here of what this next phase achieves could help creators understand and remember the distinct phases of the tutorial.

coreslinkous commented on src/tutorials/boss_tutorial.md:
```diff
@@ -0,0 +1,4330 @@
+
+For the boss to shoot a projectile, it needs a reference to the projectile template, and where to shoot the projectile from.
+
+1. Add **Shoot From** object as a custom property.
```
h4

coreslinkous commented on src/tutorials/boss_tutorial.md:
```diff
@@ -0,0 +1,4330 @@
+
+The projectile does not do any damage to the player, or the floor tiles. You need to add impact code to handle what happens to those impacted objects.
+
+1. Add `GetDamageable` function.
```
In general I would use "make" or "create" for any function that they're typing themselves, and "add" for things where they are using previously created parts.

coreslinkous commented on src/tutorials/boss_tutorial.md:
```diff
@@ -0,0 +1,4330 @@
+
+    The color property will be used to change the color of the fly up text when the boss is immune to damage.
+
+4. Add the following code.
```
More specific than code. It's fine to repeat what it is both in this title and in the explanation after.

coreslinkous commented on src/tutorials/boss_tutorial.md:
```diff
@@ -0,0 +1,4330 @@
+
+You need to listen for when the projectiles from the weapon impact an object to determine what to show for the player.
+
+1. Add `GetValidTarget` function.
```
Maybe "write".

coreslinkous commented on src/tutorials/boss_tutorial.md:
```diff
@@ -0,0 +1,4330 @@
+
+### Test the game
+
+When a player shoots the boss, it will display **Immune**.
```
specify where + picture?

coreslinkous commented on src/tutorials/boss_tutorial.md:
````diff
@@ -0,0 +1,4330 @@
+    Events.Connect("DisableWeapon", DisableWeapon)
+    ```
+
+## Add Player Sprint Ability
````
At the h2, go back to known or gerund. "Adding Player Sprint ... ", "Giving Your Players The Ability to Sprint".

It needs to sound more general than the h3 to help it break up the sections semantically.

coreslinkous commented on src/tutorials/boss_tutorial.md:
```diff
@@ -0,0 +1,4330 @@
+
+Test the game by running to each shield generator and turning it off. Make sure the following work.
+
+- Sound turns off.
```
I really love these ul's. It feels like a checklist.

coreslinkous commented on src/tutorials/boss_tutorial.md:
```diff
@@ -0,0 +1,4330 @@
+
+The **BossHealthUIClient** script needs to change the color of the various UI components that build up the heath bar, and boss name.
+
+1. Add **BossName** custom property.
```
Might be worth reinforcing custom properties with a picture of each state to check against.

coreslinkous commented on src/tutorials/boss_tutorial.md:
```diff
@@ -0,0 +1,4330 @@
+
+Share your creations with the Core Creators Discord community.
+
+<https://discord.gg/core-creators>
```
Take out Discord

CommanderFoo committed:
Updates based on review from Zaz

Most lists changed to h2/h3.
Lots more images and minor tweaks.
"Add" changed to "Create" for functions.

github-actions[bot] committed:
Optimised images with calibre/image-actions

CommanderFoo committed:
Removed comment about CC.
